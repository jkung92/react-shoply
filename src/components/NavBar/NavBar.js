import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
//import './NavBar.css';
// import styled from 'styled-components';

class NavBar extends Component {
  render() {
    const activeStyle = {
      fontWeight: 'bold',
      color: 'dodgerblue'
    };
    return (
      <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <NavLink
          className="navbar-brand"
          exact
          to="/"
          activeStyle={activeStyle}
        >
          Shoply
        </NavLink>
        <div className="navbar" id="navbarNav">
          <span className="navbar-nav">
            <li className="nav-item active">
              <NavLink
                exact
                to={`/jobs`}
                activeStyle={activeStyle}
                className="nav-link"
              >
                <p>Cart : {this.props.total}</p>
              </NavLink>
            </li>
          </span>
        </div>
      </nav>
    );
  }
}

NavBar.propTypes = {};

NavBar.defaultProps = {};

export default NavBar;
