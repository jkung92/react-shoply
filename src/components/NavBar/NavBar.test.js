import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavBar from './NavBar';
import { MemoryRouter } from 'react-router-dom';

it('renders without crashing', function() {
  shallow(<NavBar />);
});

it('matches snapshot', function() {
  const wrapper = shallow(<NavBar />);
  const serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
