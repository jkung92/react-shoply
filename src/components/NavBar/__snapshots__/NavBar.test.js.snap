// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <NavLink
    activeClassName="active"
    activeStyle={
      Object {
        "color": "dodgerblue",
        "fontWeight": "bold",
      }
    }
    aria-current="page"
    className="navbar-brand"
    exact={true}
    to="/"
  >
    Shoply
  </NavLink>
  <div
    className="navbar"
    id="navbarNav"
  >
    <span
      className="navbar-nav"
    >
      <li
        className="nav-item active"
      >
        <NavLink
          activeClassName="active"
          activeStyle={
            Object {
              "color": "dodgerblue",
              "fontWeight": "bold",
            }
          }
          aria-current="page"
          className="nav-link"
          exact={true}
          to="/jobs"
        >
          <p>
            Cart : 
          </p>
        </NavLink>
      </li>
    </span>
  </div>
</nav>
`;
